lib/log_parser.rb

```diff
@@ -1,24 +1,26 @@
+require 'json'
+
 class LogParser
   def initialize(file_path = "games.log")
     @file_name = file_path
   end
 
-  def validate_file
+  def execute
     begin   
       file = File.open(@file_name)
       process_file(file)
     rescue
       puts 'File does not exists'
```

Comment:
Lance esse puts no `STDERR` `STDERR.puts 'File does not exists'`. por padrão o ruby vai fazer o puts no `STDOUT` mas o que você quer aqui é fazer no `STDERR`. STDERR, STDOUT são streams padrões, o STDOUT é onde você manda a saída padrão da sua app, e a STDERR onde você manda a saída de erro.

Mais sobre esse tema aqui: https://rderik.com/blog/basics-of-stderr-and-stdout-on-ruby-scripts/